content/docs/command-reference/add.md

```diff
@@ -59,34 +63,28 @@ files that can be tracked with Git. See
 To avoid adding files inside a directory accidentally, you can add the
 corresponding [patterns](/doc/user-guide/dvcignore) in a `.dvcignore` file.
 
-By default DVC tries to use reflinks (see
+By default, DVC tries to use reflinks (see
 [File link types](/doc/user-guide/large-dataset-optimization#file-link-types-for-the-dvc-cache)
 to avoid copying any file contents and to optimize `.dvc` file operations for
 large files. DVC also supports other link types for use on file systems without
 `reflink` support, but they have to be specified manually. Refer to the
 `cache.type` config option in `dvc config cache` for more information.
 
-A `dvc add` target can be an individual file or a directory. There are two ways
-to work with directory hierarchies with `dvc add`:
-
-1. With `dvc add --recursive`, the hierarchy is traversed and every file is
-   added individually as described above. This means every file has its own
-   `.dvc` file, and a corresponding cached file is created (unless the
-   `--no-commit` option is used).
-2. When not using `--recursive` a `.dvc` file is created for the top of the
-   directory (with default name `dirname.dvc`). Every file in the hierarchy is
-   added to the cache (unless the `--no-commit` option is used), but DVC does
-   not produce individual `.dvc` files for each file in the directory tree.
-   Instead, the single `.dvc` file references a special JSON file in the cache
-   (with `.dir` extension), that in turn points to the files added from the
-   hierarchy.
-
-`dvc add` is typically used to version control raw data or initial datasets from
-which data processing [pipelines](/doc/command-reference/pipeline) are built,
-but it can be used to track any large file or directory. We recommend using
-`dvc run` to version control intermediate and final results (like ML models).
-This way you bring data provenance and make your project
-[reproducible](/doc/command-reference/repro).
+### Tracking directories
+
+A `dvc add` target can be an individual file or a directory. In the latter case,
+a DVC-file is created for the top of the directory (with default name
+`<dir_name>.dvc`).
+
+Every file in the hierarchy is added to the cache (unless the `--no-commit`
+option is used), but DVC does not produce individual DVC-files for each file in
+the directory tree. Instead, the single DVC-file references a special JSON file
+in the cache (with `.dir` extension), that in turn points to the added files.
+
+As an alternative, using the `--recursive` option every file is added
```

Comment:
we need to put more emphasis here that this is rarely needed, when you know what you do ... e.g. a directory with a few very large files and you want to depend on them independently ... and the preferred way to deal with dirs is still a regular `dvc add dir`

Reply:
Sure. We've done a few round on the add ref thugh, this PR is about plots mostly and other small misc. items. I'm going to leave this unresolved and extract it to a separate issue/PR if you don't mind.

Reply:
since it's already part of this PR - I would prefer to do it here.

Reply:
OK np. I gave it a shot in 7bce905b. Please take a look.